File: src/main/java/org/example/EjerJava/Ej4.java
package org.example.EjerJava;

import java.util.Random;
import java.util.Scanner;


public class Ej4 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int num = new Random().nextInt(0, 101);

        System.out.print("Adivina el número -> ");


        boolean acertado = false;
        while(!acertado) {
            String respuesta  = sc.nextLine();
            int numero = Integer.parseInt(respuesta);

            if (numero == num) {
                System.out.println("Correcto!!!");
                acertado = true;
            }else {
                System.out.print("Incorrecto!! Vuelve a intentarlo ->");
            }
        }


    }
}

File: src/main/java/org/example/ejerKotlin/Ej4.kt
package org.example.ejerKotlin

import kotlin.random.Random

fun main() {
    val num = Random.nextInt(0,101)

    print("Adivina el número -> ")

    var acertado = false

    while (!acertado){
        val numero = readln().toInt()
        if (numero == num){
            println("Correcto!!!")
            acertado = true
        }else{
            print("Número incorrecto vuelve a intentarlo -> ")
        }
    }
}